src/ProcessMaker/Nayra/Bpmn/Models/SignalEventDefinition.php

```diff
@@ -92,4 +94,32 @@ public function execute(EventDefinitionInterface $event, FlowNodeInterface $targ
     {
         return $this;
     }
+
+    /**
+     * Check if the $eventDefinition should be catch
+     *
+     * @param EventDefinitionInterface $eventDefinition
+     *
+     * @return bool
+     */
+    public function shouldCatchEventDefinition(EventDefinitionInterface $eventDefinition)
+    {
+        $targetPayload = $this->getPayload();
+        $sourcePayload = $eventDefinition->getPayload();
+        return $targetPayload && $targetPayload && $targetPayload->getId() === $sourcePayload->getId();
```

Comment:
The variable appears twice in the logic expression (i.e $targetPayload && $targetPayload)